agent/metadata_service.go

```diff
@@ -60,10 +62,17 @@ func makeSecure(handler func(http.ResponseWriter, *http.Request), mds *metadataS
 			return
 		}
 
+		parsedIP := net.ParseIP(ip)
+		allowedIP := ip == `127.0.0.1` || ip == `169.254.169.254`
+		for _, ipNet := range mds.allowIps {
+			allowedIP = allowedIP || ipNet.Contains(parsedIP)
+		}
+
 		// Must make sure the remote ip is localhost, otherwise clients on the same network segment could
 		// potentially route traffic via 169.254.169.254:80
-		if ip != `127.0.0.1` && ip != `169.254.169.254` {
-			msg := fmt.Sprintf("Access denied from non-localhost address: %s", ip)
+		if !allowedIP {
+			msg := fmt.Sprintf("Access denied from non-localhost address: %s, non-localhost allowed addresses: %v", ip, mds.allowIps)
```

Comment:
If you move the defaults as I wrote above, then:
```suggestion
			msg := fmt.Sprintf("Access denied from address: %s; not in set of allowed IPs", ip)
```